grafana-plugin/src/containers/UserSettings/parts/tabs/PersonalWebhookInfo/PersonalWebhookInfo.tsx

```diff
@@ -1,27 +1,202 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
-import { Stack } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { LoadingPlaceholder, Button, Select, Stack, Field, TextArea } from '@grafana/ui';
 import { StackSize } from 'helpers/consts';
 import { observer } from 'mobx-react';
+import { Controller, useForm } from 'react-hook-form';
 
-import { Block } from 'components/GBlock/Block';
+import { PluginLink } from 'components/PluginLink/PluginLink';
 import { Text } from 'components/Text/Text';
+import { WithConfirm } from 'components/WithConfirm/WithConfirm';
+import { ActionKey } from 'models/loader/action-keys';
+import { WebhookTriggerType } from 'models/outgoing_webhook/outgoing_webhook.types'
+import { useStore } from 'state/useStore';
+
+interface FormFields {
+  webhook: string;
+  context: string;
+}
+
+function useLoadingWebhooks() {
+  const { loaderStore } = useStore();
+  const [isLoading, setIsLoading] = React.useState(true)
+  const isLoadingWebhooks = loaderStore.isLoading(ActionKey.FETCH_WEBHOOKS)
+  const [hasRegisteredLoadingWebhooks, setHasRegisteredLoadingWebhooks] = useState(false);
+
+  useEffect(() => {
+    if (isLoadingWebhooks === true) {
+      setHasRegisteredLoadingWebhooks(true);
+    }
+  }, [isLoadingWebhooks]);
+
+  useEffect(() => {
+    if (!isLoadingWebhooks && hasRegisteredLoadingWebhooks) {
+      setIsLoading(false);
+    }
+  }, [isLoadingWebhooks])
+
+  return isLoading;
+}
+
+const defaultValues = { webhook: null, context: '{}' }
 
 export const PersonalWebhookInfo = observer(() => {
-  return (
-    <>
-      {/* allow selecting a personal webhook + setting additional metadata as context */}
-      <Stack direction="column" alignItems="flex-start" gap={StackSize.lg}>
-        <Block bordered withBackground>
-          <Stack direction="column" alignItems="center">
-            <Text>Choose a personal notification webhook</Text>
-            <br />
-            <Text>Set your user specific context data</Text>
-            <br />
-            <Text>Display current settings</Text>
-          </Stack>
-        </Block>
+  const { userStore, outgoingWebhookStore } = useStore();
+
+  const {
+    formState: {
+      isDirty,
+      isValid,
+      errors,
+    },
+    watch,
+    control,
+    getValues,
+    handleSubmit,
+    reset,
+  } = useForm<FormFields>({
+    mode: 'onBlur',
+    defaultValues
+  });
+
+  const user = userStore.items[userStore.currentUserPk];
+  const selectedWebhook = watch('webhook');
+  const hasConnectedWebhook = user.messaging_backends.WEBHOOK != null;
+  const isLoadingWebhooks = useLoadingWebhooks()
+
+  useEffect(() => {
+    (async () => {
+      await userStore.updatePersonalWebhook();
+      await outgoingWebhookStore.updateItems(
+        {
+          trigger_type: WebhookTriggerType.PersonalNotification.key,
+        },
+        true
+      );
```

Comment:
I wasn't sure about a lot of what I am doing with the stores. Is this the right thing to do here? I think I will be updating the items in the webhook store with only webhooks that are of the "personal notification" type. Could that cause issues to other pages?

Reply:
Maybe we can chat about this one, I'll ping u